backup/db.go

```diff
@@ -89,6 +90,40 @@ func (d *backupDB) setUseEncryption(use bool) error {
 	})
 }
 
+// setLatestBackupTime updates the last time the app updated.
+func (d *backupDB) setLatestBackupTime(modifiedTime string) error {
+	return d.db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(backupBucket))
+		// Check if there is an earlier modifiedTime value stored in the database.
+		if val := b.Get(latestBackupKey); val != nil {
+			// Delete the earlier modifiedTime value.
+			err := b.Delete(latestBackupKey)
+			if err != nil {
+				return err
+			}
+		}
+		// Write the new modifiedTime value to the database.
+		return b.Put(latestBackupKey, []byte(modifiedTime))
```

Comment:
One thing that concerns me is if the user has changed backup provider. It seems that things could go wrong here since the timestamp may refer to a different provider (the old one).

Reply:
So we need to specify what backup provider we're using in the db too ?

Reply:
This is one option I can think of.